app/forms/two_factor_authentication/phone_deletion_form.rb

```diff
@@ -0,0 +1,52 @@
+module TwoFactorAuthentication
+  class PhoneDeletionForm
+    include ActiveModel::Model
+
+    attr_reader :user, :configuration
+
+    def initialize(user, configuration)
+      @user = user
+      @configuration = configuration
+    end
+
+    def submit
+      success = configuration_absent? ||
+                configuration_owned_by_user? && multiple_factors_enabled? && configuration_destroyed
+
+      FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
+    end
+
+    private
+
+    def extra_analytics_attributes
+      {
+        user_id: user.uuid,
+        configuration_present: configuration.present?,
+        configuration_id: configuration&.id,
+        configuration_owner: configuration&.user&.uuid,
```

Comment:
Why are these attributes valuable to capture? Could we track the resulting configuration counts for the user after the deletion?
```ruby
mfa_method_counts: MfaContext.new(user.reload).enabled_two_factor_configuration_counts_hash
```

Reply:
The main thing is to make sure someone doesn't doctor the URL and we not catch it in the controller. The check here is just in case someone tweaks the controller and forgets to scope the configuration by user.